### DISCUSSION TITLE
Audio window size in streaming mode

### DISCUSSION THREAD

**Discussion by philn** [coqui-ai/STT#2007] [Q&A]
Hi,

I'm trying to make streaming work in a GStreamer plugin I'm working on. I'm not sure how it's supposed to work though, am I supposed to call `STT_FeedAudioContent()` with an arbitrary-sized buffer? Or does it depend on the model being used?  I think I'm after the `ModelState::audio_win_len_` field, but I don't see how to access it from the public C API.

Philippe

**Comment by reuben** [comment]
You can feed an arbitrarily sized buffer, yeah. The implementation takes care of buffering samples in between full timesteps.
